Sources/Subs-Attachments.php

```diff
@@ -1370,9 +1370,6 @@ function prepareAttachsByMsg($msgIDs)
 	else
 		$msgIDs = array_diff($msgIDs, array_keys($context['loaded_attachments']), array(0));
```

Comment:
Can we ensure on this line that we don't take in array(0) at all?  Otherwise $msgIDs could still hold id_msg of 0
```
		$msgIDs = array_diff($msgIDs, array_keys($context['loaded_attachments']));
	foreach ($msgIDs as $mID => $mVal)
		if (empty($mVal))
			unset($msgIDs[$mID]);
```

Reply:
This line already does ensure that `$msgIDs` won't contain a zero. Try this to see what I mean:

```
<?php

$msgIDs = array(123, 456, 789, 0);

$context['loaded_attachments'] = array(
	123 => array('the content of this array is irrelevant for this test'),
	789 => array('the content of this array is irrelevant for this test'),
);

$msgIDs = array_diff($msgIDs, array_keys($context['loaded_attachments']), array(0));

var_export($msgIDs);
```

Output will be:
```
array (
  1 => 456,
)
```